arithmetic/database.py

```diff
@@ -0,0 +1,173 @@
+import csv
+import os
+from contextlib import contextmanager
+
+
+class ArithmeticDatabase(object):
+    """
+    A simple database backed by a csv file to contain arithmetic operations
+    with two integers and the corresponding answer
+    """
+    headers = ['operator', 'int1', 'int2', 'answer']
+    valid_operators = ['add', 'subtract', 'multiply', 'divide']
+    buffer_size = 1024
+
+    def __init__(self, filepath, use_existing=False):
+        """
+        Initialize the csv file used as the basis for the database. Also write
+        the headers to the file
+
+        Args:
+            filepath (str): path for the csv file
+            use_existing (bool): if True, initialize with an existing file
+
+        Raises:
+            Exception: if not `use_existing` and a file with `filepath` exists
+        """
+        self.filepath = filepath
+        if os.path.exists(filepath):
+            if not use_existing:
+                return Exception('Cannot initialize ArithmeticDatabase; file %s already exists' % filepath)
+        else:
+            self.write(self.headers)
+
+    @contextmanager
+    def open_write(self):
+        """
+        Use a contextmanager to yield the csv.writer object
+
+        Yields:
+            csv.writer() object
+        """
+        csv_file = open(self.filepath, mode='a', newline='')
+        yield csv.writer(csv_file, delimiter=',')
+        csv_file.close()
+
+    @contextmanager
+    def open_read(self):
+        """
+        Use a contextmanager to yield the csv.reader object
+
+        Yields:
+            csv.writer() object
+        """
+        csv_file = open(self.filepath, mode='r+', newline='')
+        yield csv.reader(csv_file, delimiter=',')
+
+    def validate(self, row):
+        """
+        Validate the everything about the given row is correct.
+
+        Args:
+            row (list): list of elements to write in the row
+
+        Returns:
+            bool: validation status
+        """
+        if row[0] not in self.valid_operators:
```

Comment:
The specification says you're validating 'everything' about the given row, but you're only validating that the operator is valid.